### PR TITLE
Senlin: Clusters Update

### PR DESCRIPTION
**Pull request by JackKuei**
Prior to a PR being reviewed, there needs to be a Github issue that the PR
addresses. Replace the brackets and text below with that issue number.

For #[823 [Add support for Senlin clustering]](https://github.com/gophercloud/gophercloud/issues/823)

Links to the line numbers/files in the OpenStack source code that support the
code in this PR:

[[clusters:update]](https://github.com/openstack/senlin/blob/e5bc16b3fea2f10dead365f983c5a937f23d72b0/senlin/api/openstack/v1/clusters.py#L90)

### CODE REVIEW

**Comment by JackKuei** [comment]
@jtopjian Ready to be reviewed

File: openstack/clustering/v1/clusters/requests.go [hunk 1]
```diff
@@ -95,3 +95,48 @@ func List(client *gophercloud.ServiceClient, opts ListOptsBuilder) pagination.Pa
 		return ClusterPage{pagination.LinkedPageBase{PageResult: r}}
 	})
 }
+
+// UpdateOpts implements UpdateOpts
+type UpdateOpts struct {
+	Config      string                 `json:"config,omitempty"`
+	Name        string                 `json:"name,omitempty"`
+	ProfileID   string                 `json:"profile_id,omitempty"`
+	Timeout     int                    `json:"timeout,omitempty"`
```
Comment (jtopjian) [inline openstack/clustering/v1/clusters/requests.go]: Is a timeout value of `0` valid? If so then this should be `*int`.

Comment (JackKuei) [reply openstack/clustering/v1/clusters/requests.go]: Should allow a Timeout of 0. Made changes

**Comment by JackKuei** [comment]
@jtopjian Changed Timeout from int to *int to allow value of 0

**Comment by JackKuei** [comment]
Ready to be reviewed

**Comment by jtopjian** [comment]
@JackKuei apologies for the delay. Thank you 😄

**Pull request merged**